lib/tty/prompt/multi_list.rb

```diff
@@ -62,6 +63,7 @@ def keyspace(*)
         else
           return if @max && @selected.size >= @max
           @selected << active_choice
+          @selected.sort_by! { |choice| choices.index(choice) } if @preserve_order
```

Comment:
To me, this should be the default behaviour. I wonder if we could extend the `Choice` with `index` attribute or similar and make it comparable. This would simplify and speed up sorting:

```ruby
@selected.sort!
``` 

Alternatively, we could extend the `Choice` with a new state like `selected`. This would have an added benefit that you could define your choice as selected before the menu is rendered. Then we could remove the `@selected` collection and replace the call with an already sorted list:

```ruby
choices.map(&:selected)
```

Reply:
In the last code sample did you perhaps mean `choices.select(&:selected?)` (`Enumerable#select`, not `Enumerable#map`)? I love this option, if you don't mind I'll try exploring it righ in this PR to check whether it works as good as it sounds.

Reply:
Good spot - amended my example. This will impact how the `select` prompt works as well. The two prompts share some functionality, notably, the `selected` collection. It's worth exploring as this would make API more flexible, readable and improve performance.